lib/Map/Vector/Protomaps/tableStyleToProtomaps.ts

```diff
@@ -0,0 +1,262 @@
+import { runInAction } from "mobx";
+import {
+  CircleSymbolizer,
+  Feature,
+  GeomType,
+  JsonValue,
+  LabelRule,
+  LineSymbolizer,
+  PaintRule,
+  PolygonSymbolizer,
+  Feature as ProtomapsFeature
+} from "protomaps-leaflet";
+import TimeInterval from "terriajs-cesium/Source/Core/TimeInterval";
+import { isJsonNumber } from "../../../Core/Json";
+import { NotNull, NotUndefined } from "../../../Core/TypeModifiers";
+import GeoJsonMixin, {
+  FEATURE_ID_PROP,
+  getStyleReactiveDependencies
+} from "../../../ModelMixins/GeojsonMixin";
+import TableStyleMap, {
+  isConstantStyleMap
+} from "../../../Table/TableStyleMap";
+import { TableStyleMapSymbolTraits } from "../../../Traits/TraitsClasses/Table/StyleMapTraits";
+import { GEOJSON_SOURCE_LAYER_NAME } from "./ProtomapsGeojsonSource";
+
+/** Helper function to create a function which can be used to generate styles for a given rowID. This ensures that nothing is read outside reactive context */
+function getStyleValueFn<
+  T extends TableStyleMapSymbolTraits,
+  Key extends keyof T,
+  Value extends NotUndefined<NotNull<T[keyof T]>>
+>(
+  tableStyleMap: TableStyleMap<T>,
+  trait: Key,
+  defaultValue: Value,
+  useRowId = true,
+  transformValue?: (v: Value) => Value
+): (z: number, f?: ProtomapsFeature) => Value {
+  const styleMap = runInAction(() => tableStyleMap.styleMap);
+
+  if (isConstantStyleMap(styleMap)) {
+    return () => {
```

Comment:
Why does TypeScript not flag this as a type error, isn't the return type a function that takes a number?

Reply:
It is valid JS/TS to omit arguments.